lib/kubernetes-deploy/kubectl_wrapper.rb

```diff
@@ -0,0 +1,21 @@
+module KubernetesDeploy
+  module KubectlWrapper
+    extend self
+
+    def run_kubectl(*args, context: nil, namespace: nil)
+      args = args.unshift("kubectl")
+      args.push("--context=#{context}") if context
+      args.push("--namespace=#{namespace}") if namespace
+
+      KubernetesDeploy.logger.debug Shellwords.join(args)
+      out, err, st = Open3.capture3(*args)
+      KubernetesDeploy.logger.debug(out.shellescape)
+
+      unless st.success?
```

Comment:
This will spew irrelevant warnings when the polling code uses it. For example, the return code on all 404s before the resources come up is not success. As I explained on the issue you linked to in the description:

> The original reason for having two different versions is a difference in error handling and logging. The runner version handles critical deploy commands that could fail, and both its callers and the end user should be interested in error output as a result. The resources on the other hand are doing polling, in which context typical !st.success? causes are transient or even expected, e.g. 404s on a GETs for resources that aren't up yet.